### ISSUE TITLE
Crash when reconfiguring local config to start exam config

### ISSUE THREAD

**Issue by abowman-unige** [SafeExamBrowser/seb-win-refactoring#607]
Hello,

We have some random crash when we reconfigure SEB from the local configuration to a start exam configuration. It seems to appear about 5 to 10 % of the time. Is there a problem while stopping and restarting the service ?

- OS: Windows 10 
- SEB: 3.5
- Virtual machine with double screen and seb autostart through GPO

[reconfig-crash-v35.zip](https://github.com/SafeExamBrowser/seb-win-refactoring/files/11380948/reconfig-crash-v35.zip)

Thanks.

**Comment by dbuechel** [comment]
Thanks for your report. We'll have a look at the logs and report back here.

**Comment by dbuechel** [comment]
Yes, there appears to be an issue in the service component when starting the new session:

```
2023-05-03 09:09:39.971 [03] - ERROR: Caught unexpected exception while performing operation 'LockdownOperation'!

   Exception Message: La référence d'objet n'est pas définie à une instance d'un objet.
   Exception Type: System.NullReferenceException

   à SafeExamBrowser.Service.Operations.LockdownOperation.Perform() dans C:\Users\appveyor\projects\seb-win-refactoring-ist041kp6jai2xxx\SafeExamBrowser.Service\Operations\LockdownOperation.cs:ligne 43
   à SafeExamBrowser.Core.OperationModel.OperationSequence.Perform() dans C:\Users\appveyor\projects\seb-win-refactoring-ist041kp6jai2xxx\SafeExamBrowser.Core\OperationModel\OperationSequence.cs:ligne 108

```

https://github.com/SafeExamBrowser/seb-win-refactoring/blob/master/SafeExamBrowser.Service/Operations/LockdownOperation.cs#L43

However, the error message does not really make sense, as there cannot be a NullReferenceException on that line of code. Do you happen to have a reliable way to reproduce the problem or could you elaborate a bit how exactly you're starting, using and then reconfiguring SEB when the issue occurs?

**Comment by dbuechel** [comment]
Thanks a lot for the additional information. I'll try to have a look as soon as the development schedule and other priorities allow.

**Comment by dbuechel** [comment]
I have noticed a flickering in the video you posted (one especially strange at around 1:27, where SEB should already be terminated but then the SEB desktop reappears for a second); I do assume that this is an issue of the screen recording rather than an actual flickering of SEB's user interface?

Apart from that, I am unfortunately perplexed by the issue you're having. As already stated above, that particular line in the lockdown operation of the service cannot produce a `NullReferenceException` under normal circumstances. The only explanation I could hazard would be that there is something fundamentally wrong with either the system itself or a component SEB relies upon to work properly (e.g. the .NET Framework).

Thus, unless you'd have a reliable way to reproduce the issue, I'm afraid that I can't do much about the issue itself. I do know from our setup that our system administrators have put a 30 second timeout before actually starting SEB, but it would be spectacularly strange if that'd have any effect on the problem at hand.

**Comment by RubyBit** [comment]
I am not knowledgeable on the codebase, but I am interested in the project and reading through the issues..

Maybe the line referencing `NullReferenceException` on the stack trace is wrong, that is because there could have been a compilations optimizations or some dll injection which pollute the call stack and mess up the lines (offsets). Maybe this is even cause by the reconfiguration or because the vm itself is injecting to the process (specially in conjuction to a multi threaded context?). The vm could also not have all the appropriate services running (unlikely though because the error is intermittent).

Just an uninformed shot in the dark but could be helpful.

**Comment by dbuechel** [comment]
Thanks for the input, yes it is indeed very strange, I can't quite get my head around it. This is actually the first time that a stack trace reported by users is "off", up and until now the line of failure always was accurate.

@abowman-unige Do you have any of the optimizations et al. mentioned by @RubyBit in place in your virtualization setup?

**Comment by dbuechel** [comment]
Thanks a lot for the update @abowman-unige. This indeed would make a lot of sense and does also coincide with experiences we made ourselves with our virtualization setup, where we realized that virtual machines / desktops need to be sufficiently potent in order to run smoothly and without similar, seemingly random issues like this one.

It is not clear to me from your comments how big the actual system load was in your situation, but I could imagine that the underlying issue here was that Windows itself reached limits and started to terminate running tasks in order to keep important resources free for the OS to complete its basic functionality.
